### ISSUE TITLE
Discovery: Confirm the shape of AC-API error response (and tkt bug fix if needed)

### ISSUE THREAD

**Issue by kjsuarez** [department-of-veterans-affairs/va.gov-team#99724]
## Description
Confirm via datadog that the AC-API error responses in production match the docs.
### User story

As a Cartography team member, I want to be certain that we are correctly consuming the error response from the AC-API

### Notes
- Wes asked Trevor Bosaw and got lots of good stuff: [slack](https://dsva.slack.com/archives/C04LFGY2SJK/p1735605092344189?thread_ts=1735603760.781789&cid=C04LFGY2SJK)

Possible tasks:
- find a log example of an AC-api response
- if shape is wrong, ticket bug fix

**Comment by wesrowe** [comment]
Trevor provided [some good stuff](https://dsva.slack.com/archives/C04LFGY2SJK/p1735605092344189?thread_ts=1735603760.781789&cid=C04LFGY2SJK).

**Comment by kjsuarez** [comment]
Daniel and I did some digging in datadog and it looks like the error response we're getting may not include a status code in the place we expect to get one, which would explain why we're not seeing any user-actionable errors. I've reached out to Trevor to confirm. Discussion ongoing.

**Comment by wesrowe** [comment]
@kjsuarez, wearing my Mark hat, could you please link that slack thread to keep the ticket complete?

**Comment by kjsuarez** [comment]
https://dsva.slack.com/archives/C04LFGY2SJK/p1735603760781789

**Comment by wesrowe** [comment]
@kjsuarez, I want to make sure we're on the path to getting this ticket unblocked. Looking at the thread you linked above, it's not obvious that Identity has an action item to resolve the missing error code field. If it's ambiguous we can check in about this in standup Wed, I'll do whatever is needed to get us unblocked.

**Comment by dcloud** [comment]
The [Route guard aka "useMyHealthAccessGuard"](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/platform/mhv/hooks/useMyHealthAccessGuard.jsx) that checks `mhvAccountState`

**Comment by wesrowe** [comment]
Notes from Trevor/Joe meeting: 
- mhv_user_accounts_controller.rb is where the action is
- Datadog query for all client errors for MHV Account Creation Service (similar query to one in dash): https://vagov.ddog-gov.com/logs?query=%22%5BMHV%5D%5BAccountCreation%5D%5BService%5D%22%20create_account%20-%40message_content%3A%28%22%5BMHV%5D%5BAccountCreation%5D%5BService%5D%20create_account%20success%22%20OR%20%22%5BMHV%5D%5BAccountCreation%5D%5BService%5D%20create_account%20request%22%29&agg_m=count&agg_m_source=base&agg_t=count&clustering_pattern_field_path=message&cols=host%2Cservice&fromUser=true&messageDisplay=inline&refresh_mode=sliding&storage=flex_tier&stream_sort=desc&viz=stream&from_ts=1736371158591&to_ts=1736372058591&live=true
- "validation errors" indicate controller didn't have sufficient data to call API at all; it never calls it (e.g., missing ICN)
- found in an example of an 805 error referred by /my-health not getting sent to FE - Trevor will investigate by end of week
- landing page has a bug, shouldn't call ACapi for unverified users

**Comment by kjsuarez** [comment]
Results of this ticket:
- It is now demonstrably apparent that AC-API is sending us error responses without  the needed `code` attribute, thus causing the lack of user-actionable errors to surface on GA. Identity team has agreed to address the issue.
- [100539](https://github.com/department-of-veterans-affairs/va.gov-team/issues/100539) drafted to address a separate issue in which we call AC-API unnecessarily
- [100524](https://github.com/department-of-veterans-affairs/va.gov-team/issues/100524) drafted to handle error response without `code` attribute

**Comment by kjsuarez** [comment]
[PR](https://github.com/department-of-veterans-affairs/vets-website/pull/34053)

**Comment by mcarterA6** [comment]
@kjsuarez - this ticket was initially scoped for 1 point. please update points and add a note to capture the change the point change if appropriate.

**Comment by kjsuarez** [comment]
Re-scoped from a 1 to a 3. This work includes:
- Datadog access
- tracking AC-API logs to confirm the missing attribute
- communicating with Identity to highlight the issue
- confirming the change in API response schema
- PR to adjust code to account for change in schema
